### PR TITLE
crimson/seastore: minor fix for seastore debugging

### PR DESCRIPTION
**Pull request by wuxingyi**
1、 seastore deserves a logging subsystem now
2、use `__func__` for logging in non-lamda functions

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/common/subsys.h [hunk 1]
```diff
@@ -67,6 +67,7 @@ SUBSYS(throttle, 1, 1)
 SUBSYS(refs, 0, 0)
 SUBSYS(compressor, 1, 5)
 SUBSYS(bluestore, 1, 5)
+SUBSYS(seastore, 1, 5)
```
Comment (tchaikov) [inline src/common/subsys.h:70]: might want to guard this line with `#ifdef WITH_SEASTAR`.

**Review by tchaikov — COMMENTED** [review]
i don't feel strong if we should use filestore or seastore.  will defer this to @athanatos

File: src/common/subsys.h (hunk 1 above)
Comment (athanatos) [reply src/common/subsys.h:70]: I think this is fine without the ifdef.

File: src/crimson/os/seastore/extentmap_manager/btree/extentmap_btree_node_impl.cc [hunk 2]
```diff
@@ -266,7 +266,8 @@ ExtMapLeafNode::insert(ext_context_t ec, objaddr_t lo, lext_map_val_t val)
   journal_insert(insert_pt, lo, val, maybe_get_delta_buffer());
 
   logger().debug(
-    "ExtMapLeafNode::insert: inserted {}->{} {}",
+    "{}: inserted {}->{} {}",
```
Comment (athanatos) [inline src/crimson/os/seastore/extentmap_manager/btree/extentmap_btree_node_impl.cc:269]: I actually want the class name here.

File: src/crimson/os/seastore/extentmap_manager/btree/extentmap_btree_node_impl.cc [hunk 3]
```diff
@@ -248,7 +248,7 @@ ExtMapLeafNode::find_lextent(ext_context_t ec, objaddr_t lo, extent_len_t len)
       (*from).get_key(),
       val.laddr,
       val.length));
-    logger().debug("ExtMapLeafNode::find_lextent find {}~{}", lo, val.laddr);
```
Comment (athanatos) [inline src/crimson/os/seastore/extentmap_manager/btree/extentmap_btree_node_impl.cc:251]: I actually want the class name here.

File: src/crimson/os/seastore/extentmap_manager/btree/extentmap_btree_node_impl.cc [hunk 4]
```diff
@@ -236,7 +236,7 @@ ExtMapLeafNode::find_lextent_ret
 ExtMapLeafNode::find_lextent(ext_context_t ec, objaddr_t lo, extent_len_t len)
 {
   logger().debug(
-    "ExtMapLeafNode::find_lextent {}~{}", lo, len);
```
Comment (athanatos) [inline src/crimson/os/seastore/extentmap_manager/btree/extentmap_btree_node_impl.cc:239]: I actually want the class name here.

File: src/crimson/os/seastore/extentmap_manager/btree/extentmap_btree_node_impl.cc [hunk 5]
```diff
@@ -163,7 +163,7 @@ ExtMapInnerNode::merge_entry(ext_context_t ec, objaddr_t lo,
     auto mut_iter = mut->iter_idx(iter->get_offset());
     return mut->merge_entry(ec, lo, mut_iter, entry);
   }
-  logger().debug("ExtMapInnerNode: merge_entry: {}, {}", *this, *entry);
```
Comment (athanatos) [inline src/crimson/os/seastore/extentmap_manager/btree/extentmap_btree_node_impl.cc:166]: I actually want the class name here.

File: src/crimson/os/seastore/extentmap_manager/btree/extentmap_btree_node_impl.cc [hunk 6]
```diff
@@ -287,7 +288,8 @@ ExtMapLeafNode::rm_lextent(ext_context_t ec, objaddr_t lo, lext_map_val_t val)
            && val.length == rm_pt->get_val().length) {
     journal_remove(rm_pt, maybe_get_delta_buffer());
     logger().debug(
-      "ExtMapLeafNode::rm_lextent: removed {}->{} {}",
```
Comment (athanatos) [inline src/crimson/os/seastore/extentmap_manager/btree/extentmap_btree_node_impl.cc:290]: I actually want the class name here.

File: src/crimson/os/seastore/lba_manager/btree/lba_btree_node_impl.cc [hunk 7]
```diff
@@ -185,8 +185,8 @@ LBAInternalNode::find_hole_ret LBAInternalNode::find_hole(
   extent_len_t len)
 {
   logger().debug(
-    "LBAInternalNode::find_hole min={}, max={}, len={}, *this={}",
```
Comment (athanatos) [inline src/crimson/os/seastore/lba_manager/btree/lba_btree_node_impl.cc:188]: I actually want the class name here.

File: src/crimson/os/seastore/lba_manager/btree/lba_btree_node_impl.cc [hunk 8]
```diff
@@ -302,7 +303,8 @@ LBAInternalNode::split_entry(
   c.cache.retire_extent(c.trans, entry);
 
   logger().debug(
-    "LBAInternalNode::split_entry *this {} entry {} into left {} right {}",
```
Comment (athanatos) [inline src/crimson/os/seastore/lba_manager/btree/lba_btree_node_impl.cc:305]: I actually want the class name here.

File: src/crimson/os/seastore/lba_manager/btree/lba_btree_node_impl.cc [hunk 9]
```diff
@@ -326,7 +328,8 @@ LBAInternalNode::merge_entry(
   }
 
   logger().debug(
-    "LBAInternalNode: merge_entry: {}, {}",
```
Comment (athanatos) [inline src/crimson/os/seastore/lba_manager/btree/lba_btree_node_impl.cc:329]: I actually want the class name here.

File: src/crimson/os/seastore/lba_manager/btree/lba_btree_node_impl.cc [hunk 10]
```diff
@@ -411,7 +414,8 @@ LBALeafNode::lookup_range_ret LBALeafNode::lookup_range(
   extent_len_t len)
 {
   logger().debug(
-    "LBALeafNode::lookup_range {}~{}",
```
Comment (athanatos) [inline src/crimson/os/seastore/lba_manager/btree/lba_btree_node_impl.cc:414]: I actually want the class name here.

File: src/crimson/os/seastore/lba_manager/btree/lba_btree_node_impl.cc [hunk 11]
```diff
@@ -443,7 +447,8 @@ LBALeafNode::insert_ret LBALeafNode::insert(
 
   val.paddr = maybe_generate_relative(val.paddr);
   logger().debug(
-    "LBALeafNode::insert: inserting {}~{} -> {}",
```
Comment (athanatos) [inline src/crimson/os/seastore/lba_manager/btree/lba_btree_node_impl.cc:446]: I actually want the class name here.

File: src/crimson/os/seastore/lba_manager/btree/lba_btree_node_impl.cc [hunk 12]
```diff
@@ -452,7 +457,8 @@ LBALeafNode::insert_ret LBALeafNode::insert(
   journal_insert(insert_pt, laddr, val, maybe_get_delta_buffer());
 
   logger().debug(
-    "LBALeafNode::insert: inserted {}~{} -> {}",
```
Comment (athanatos) [inline src/crimson/os/seastore/lba_manager/btree/lba_btree_node_impl.cc:455]: I actually want the class name here.

File: src/crimson/os/seastore/lba_manager/btree/lba_btree_node_impl.cc [hunk 13]
```diff
@@ -527,8 +533,8 @@ LBALeafNode::find_hole_ret LBALeafNode::find_hole(
   extent_len_t len)
 {
   logger().debug(
-    "LBALeafNode::find_hole min={} max={}, len={}, *this={}",
```
Comment (athanatos) [inline src/crimson/os/seastore/lba_manager/btree/lba_btree_node_impl.cc:530]: I actually want the class name here.

File: src/crimson/os/seastore/lba_manager/btree/lba_btree_node_impl.cc [hunk 14]
```diff
@@ -585,7 +591,8 @@ void LBALeafNode::resolve_relative_addrs(paddr_t base)
       auto val = i->get_val();
       val.paddr = base.add_relative(val.paddr);
       logger().debug(
-	"LBALeafNode::resolve_relative_addrs {} -> {}",
```
Comment (athanatos) [inline src/crimson/os/seastore/lba_manager/btree/lba_btree_node_impl.cc:588]: I actually want the class name here.

**Comment by athanatos** [comment]
I like swapping the logger, but I'd like to leave the ClassName:: prefixes where they exist.

**Comment by wuxingyi** [comment]
jenkins retest this please

File: src/crimson/os/seastore/cache.cc [hunk 15]
```diff
@@ -168,7 +168,7 @@ CachedExtentRef Cache::duplicate_for_write(
 
   ret->version++;
   ret->state = CachedExtent::extent_state_t::MUTATION_PENDING;
-  logger().debug("Cache::duplicate_for_write: {} -> {}", *i, *ret);
+  logger().debug("{}: {} -> {}", __func__, *i, *ret);
```
Comment (athanatos) [inline src/crimson/os/seastore/cache.cc:171]: This removes the Cache:: prefix.

File: src/crimson/os/seastore/cache.cc [hunk 16]
```diff
@@ -440,8 +440,7 @@ Cache::get_next_dirty_extents_ret Cache::get_next_dirty_extents(
 	ret,
 	[](auto &ext) {
 	  logger().debug(
-	    "get_next_dirty_extents: waiting on {}",
-	    *ext);
+	    "get_next_dirty_extents: waiting on {}", *ext);
```
Comment (athanatos) [inline src/crimson/os/seastore/cache.cc:443]: This doesn't seem meaningful.

File: src/crimson/os/seastore/journal.cc [hunk 17]
```diff
@@ -476,7 +477,7 @@ Journal::scan_segment_ret Journal::scan_segment(
   delta_scan_handler_t *delta_handler,
   extent_handler_t *extent_info_handler)
 {
-  logger().debug("Journal::scan_segment: starting at {}", addr);
+  logger().debug("{}: starting at {}", __func__, addr);
```
Comment (athanatos) [inline src/crimson/os/seastore/journal.cc:480]: I want to keep the Journal:: prefix.

**Comment by athanatos** [comment]
There are still a bunch of class name prefixes being removed.

**Comment by tchaikov** [comment]
@wuxingyi why did you close this PR?

**Pull request closed without merging**